### DISCUSSION TITLE
FoxEss Upgrade

### DISCUSSION THREAD

**Discussion by gjhill6100** [nathanmarlor/foxess_modbus#358] [Q&A] [answered]
Hi,

Can you upgrade to the latest FoxESS_Modbus version and still keep the history of the previous registers, historical values, etc ?

Do you just delete the previous integration and install the newer version ?

Will the historical information be preserved if you keep the same Inverter Name ?

I have also created a couple of Template Sensors in Config.yaml and assume that they will all still work as long as the name of the sensor created in FoxEss_Modbus was to stay the same ?

Regards for any guidance

**Comment by canton7** [comment]
Just upgrade using HACS, no need to remove / reinstall. But even if you do remove / reinstall your sensor history is preserved, provided you keep the same entity id prefix. Template sensors aren't affected by upgrades at all.

**Reply by gjhill6100** [reply]
HACS is not showing me any options to update FoxESS_Modbus unless I have managed to confuse it be downloading the updated Version.
![Screenshot 2023-07-03 132923](https://github.com/nathanmarlor/foxess_modbus/assets/120195446/b70561a9-170d-4a50-8abb-9f0fd0506d8f)

**Reply by canton7** [reply]
Weird, click on it, see what version it thinks you have installed? There are some other options under the 3 dots in the top right (after you've clicked on it) as well

**Reply by gjhill6100** [reply, marked as answer]
Looks like 1.6.2 so maybe I did something without knowing it :)

Thanks again.

**Reply by canton7** [reply]
Cool!
